docs: improve benchmarking results for istio access logs (new provider) (#2088)
kyma-project/telemetry-manager: docs/contributor/benchmarks/istio-access-logs-otlp-provider.md

# Istio Access Logs OTLP Provider Load Test
Goal: Ensure that enabling Istio access logs with the new OTLP provider `kyma-logs` does not negatively impact Envoy's performance.

## Setup

### Base Setup
1. `istio` module deployed

> 2. Change `kyma-system/istio-controller-manager` Deployment image to: `europe-central2-docker.pkg.dev/sap-se-cx-kyma-goat/istio/istio-manager:PR-1374` (Alternatively, scale down the `istio-controller-manager` Deployment and apply [istio-config.yaml](./assets/istio-access-logs/istio-config.yaml) for additional, custom provider configurations)

Reviewer: Does this mean the alternative method provides more config options than the first method? In that case, I'd suggest

```suggestion
2. Change `kyma-system/istio-controller-manager` Deployment image to: `europe-central2-docker.pkg.dev/sap-se-cx-kyma-goat/istio/istio-manager:PR-1374`.
Alternatively, if you need additional, custom provider configurations, scale down the `istio-controller-manager` Deployment and apply [istio-config.yaml](./assets/istio-access-logs/istio-config.yaml).
```

# Istio Access Logs OTLP Provider Load Test

> Goal: Ensure that enabling Istio access logs via the new OTLP provider `kyma-logs` does not negatively impact Envoy's performance.

Reviewer: ```suggestion
Goal: Ensure that enabling Istio access logs with the new OTLP provider `kyma-logs` does not negatively impact Envoy's performance.
```

2025-04-23T09:47:31.271039Z    stream closed EOF for load-test/traffic-generator (fortio)
```

> ### 📊 Scenario-based Filtered Results

Reviewer: ```suggestion
### 📊 Scenario-Based Filtered Results
```

| :---: | :--------------: | :-------------------------------: | :--------------------: | :---------------------------: | :----------------------------: | :-----------------------------------------: | :---------------------------------------: | :--------------------------------------------: | :----------------------------------: | :---------------------------: | :-------------------------------------------: |
|  R01  |    kyma-logs     |            Functional             |          504           |             2.36              |           504 / 504            |                 439 / 1230                  |               566 / 665 p/s               |                     0 p/s                      |   istio-proxy: 0.249, nginx: 0.061   |       istio-proxy: 100%       |    istio-proxy: 44.8 MiB, nginx: 4.47 MiB     |

> |  R02  | telemetry-stdout |            Functional             |          564           |             1.95              |           564 / 564            |                  485 / 827                  |               566 / 649 p/s               |                     0 p/s                      |   istio-proxy: 0.250, nginx: 0.063   |       istio-proxy: 100%       |    istio-proxy: 49.8 MiB, nginx: 4.48 MiB     |

Reviewer 2: The old provider is called "stdout-json", the term used by you is confusing. please check the whole document for occurences.

Author: You are right, idk why I had it named that way in this doc.

|  R03  |    kyma-logs     |       Backend not reachable       |          466           |             6.34              |           466 / 466            |                  818 / 421                  |               548 / 480 p/s               |                     0 p/s                      |   istio-proxy: 0.17, fortio: 0.0528   |        istio-proxy: 0%        |    istio-proxy: 40.5 MiB, fortio: 10.7 MiB     |
|  R04  |    kyma-logs     | Backend refusing some access logs |          522           |              6.8              |           522 / 522            |                  748 / 459                  |               605 / 525 p/s               |                     0 p/s                      |   istio-proxy: 0.17, fortio: 0.0509   |        istio-proxy: 0%        |    istio-proxy: 40.4 MiB, fortio: 11.4 MiB     |

> > Mentions:

Reviewer: ```suggestion
> **NOTE:**
```

> > - When not other specified, the observed metrics are reported as a rate (per second) over a 10 min timerange.

Reviewer: ```suggestion
> - Unless specified otherwise, the observed metrics are reported as a rate (per second) over a 10 min time range.
```

> > - "[Istio] ..." metrics are reported strictly in the context of the `fortio <-> nginx` traffic. Where the values reported in the `nginx` table refer to the `istio-proxy` instance of the `nginx` Pod, and the ones in the `fortio` table refer to the one of the `fortio` Pod (since these are exported as separate metrics, with the `instance` attribute set accordingly).

Reviewer: ```suggestion
> - "[Istio] ..." metrics are reported strictly in the context of the `fortio <-> nginx` traffic. The values reported in the `nginx` table refer to the `istio-proxy` instance of the `nginx` Pod, and the values in the `fortio` table refer to the one of the `fortio` Pod (because these are exported as separate metrics, with the `instance` attribute set accordingly).
```

Not quite sure what you mean with "where". I believe we can drop it?

Author: I just wanted to reference the previous sentence, but I guess you're right, looks cleaner with your version.

> > - "[Istio] Request Duration (ms)" is reported as sum/count (i.e. `istio_request_duration_milliseconds_sum` / `istio_request_duration_milliseconds_count`) over the 10 min timerange.

Reviewer: ```suggestion
> - "[Istio] Request Duration (ms)" is reported as sum/count (that is, `istio_request_duration_milliseconds_sum` / `istio_request_duration_milliseconds_count`) over the 10 min time range.
```

Author: Should I generally avoid using "i.e.", I'm quite accustomed to it? 🤔

Reviewer: That's correct - we avoid all such [Latin abbreviations](https://help.sap.com/docs/SNG_FOR_UA/e33c591ae4494a659a3f5f983c9d1161/0511266eabeb4067b395b55e41280f26.html?locale=en-US&state=DRAFT&q=i.e.#english-no-latin-abbreviations) :)

Author: I did not knew that, thank you!

Observed behavior:
  - No hint of dropped access logs or any other erroneous behavior was observed.

>   - Access logs were not reaching the backend, but most (see next point) of the envoy/istio metrics did not report this behavior.

Reviewer: ```suggestion
  - Access logs did not reach the backend, but most (see next point) of the Envoy/Istio metrics did not report this behavior.
```

memorylimiter@v0.123.0/memorylimiter.go:212    Memory usage is above soft limit. Refusing data.
```
- Backpressure was not backpropagated, with the istio-proxy traffic being virtually unaffected.
- One of the places where a difference was still perceived, was in the `istio_request_messages_total / istio_response_messages_total` metric, where a value of `29.5 / 5.15` was reported.
- Resource consumption did not change compared to the normal working scenario.

## Conclusion

> Comparing the old provider (`envoy` and `telemetry-stdout`) with the new provider (`kyma-logs`), no significant differences were observed in terms of resource consumption and performance. The new provider (`kyma-logs`) seems to be able to handle the same amount of traffic as the old provider (`envoy` and `telemetry-stdout`), with similar CPU and memory usage and a slight increase in network bandwidth usage.

Reviewer 2: It is much better no with the comparing table and using the istio metrics. However, the [Istio] Requests Total metric indicates a performance degradation which we cannot neglect and need to discuss. You are building the pure average over the 10 minutes.. can you check if that metric is really that nicely averageable, are there no bigger spikes? Also, when you repeat the test several times, how big is the variation of that value, is there a significant spreading factor?
If there is really such impact, can you maybe drop all log attributes of the provider config to see if the degradation is caused by the CEL expressions or by the OTLP communication?

Observed behavior:
  - No hint of dropped access logs or any other erroneous behavior was observed.
  - Access logs did not reach the backend, but most (see next point) of the Envoy/Istio metrics did not report this behavior.

>   - One of the places where a difference was still perceived, was in the `istio_request_messages_total / istio_response_messages_total` metric, where a value of `0 / 0` was reported.

Reviewer 2: but not for the fortio - nginx traffic, right?

Author: Yes, not for that one

memorylimiter@v0.123.0/memorylimiter.go:212    Memory usage is above soft limit. Refusing data.
```
- Backpressure was not backpropagated, with the istio-proxy traffic being virtually unaffected.

> - One of the places where a difference was still perceived, was in the `istio_request_messages_total / istio_response_messages_total` metric, where a value of `29.5 / 5.15` was reported.

Reviewer 2: Same as above, that where istio metrics of the backend?

Author: Yes, will make it clearer

- Resource consumption did not change compared to the normal working scenario.

## Conclusion

> Comparing the old provider (`envoy` and `stdout-json`) with the new provider (`kyma-logs`), no significant differences were observed in terms of resource consumption and performance. The new provider (`kyma-logs`) seems to be able to handle the same amount of traffic as the old provider (`envoy` and `stdout-json`), with similar CPU and memory usage and a slight increase in network bandwidth usage.

Reviewer 2: Maybe add a remark to use the reduced label set
